backends/hif/pk-backend-hif.c

```diff
@@ -1171,30 +1172,20 @@ pk_backend_get_updates (PkBackend *backend,
 	pk_backend_job_thread_create (job, pk_backend_search_thread, NULL, NULL);
 }
 
-/* Obviously hardcoded and Fedora specific.  Colin Walters thinks this
- * concept should be based on user's trust of a GPG key or something
- * more flexible.
+/* Obviously hardcoded based on the repository prefixes compiled in.
+ * Colin Walters thinks this concept should be based on
+ * user's trust of a GPG key or something more flexible.
  */
 static gboolean
 source_is_supported (HifSource *source)
 {
 	const char *id = hif_source_get_id (source);
 	guint i;
-	const gchar *valid[] = { "fedora",
-				 "fedora-debuginfo",
-				 "fedora-source",
-				 "rawhide",
-				 "rawhide-debuginfo",
-				 "rawhide-source",
-				 "updates",
-				 "updates-debuginfo",
-				 "updates-source",
-				 "updates-testing",
-				 "updates-testing-debuginfo",
-				 "updates-testing-source",
-				 NULL };
+
+	const gchar **valid = hif_get_supported_source_prefixes();
+
 	for (i = 0; valid[i] != NULL; i++) {
-		if (g_strcmp0 (id, valid[i]) == 0)
+		if (g_str_has_prefix (id, valid[i]) == TRUE)
```

Comment:
A random drive-by comment: not sure it's a good idea to use g_str_has_prefix() to match repo names; it could very easily lead to matching something like "updates-rpmfusion" as a valid Fedora repo.